src/Controller/Controller.php

```diff
@@ -714,6 +714,191 @@ public function isAction($action)
+     * ```
+     *
+     * Accepts simplified representation of callables:
+     * ```
+     * // assuming that 'init' and 'condition' are methods on current controller
+     * // you can write
+     * $this->addBeforeAction('init', ['if' => 'condition'])
+     * // instead of
+     * $this->addBeforeAction([$this, 'init'], ['if' => [$this, 'condition']])
+     * ```
+     *
+     * @param callable|string $callable Callable to be invoked as before action (before filter)
+     * @param array $options Array of options
+     * @return void
+     */
+    public function addBeforeAction($callable, $options = [])
+    {
+        $this->on('Controller.startup', $callable, $options);
+    }
+
+    /**
+     * Adds a function that will be called after the controller action is run and rendered.
+     *
+     * You can configure the function to only execute after some of the controller actions:
+     * ```
+     * $this->addAfterAction(function () { ... }, ['only' => 'index']);
+     * $this->addAfterAction(function () { ... }, ['only' => ['index', 'view']]);
+     * ```
+     *
+     * Or you can exclude some controller actions from executing the function:
+     * ```
+     * $this->addAfterAction(function () { ... }, ['except' => 'index']);
+     * ```
+     *
+     * Options may also contain conditions and the function only executes if they all pass:
+     * ```
+     * $this->addAfterAction(function () { ... }, [
+     *     'if' => function () {
+     *         return !is_null($this->currentUser);
+     *     }
+     * ]);
+     *
+     * $this->addAfterAction(function () { ... }, [
+     *     'unless' => function () {
+     *         return is_null($this->currentUser);
+     *     }
+     * ]);
+     *
+     * $this->addAfterAction(function () { ... }, [
+     *     'if' => [
+     *          function () { ... },
+     *          function () { ... },
+     *     ]
+     * ]);
+     * ```
+     *
+     * Accepts simplified representation of callables:
+     * ```
+     * // assuming that 'init' and 'condition' are methods on current controller
+     * // you can write
+     * $this->addAfterAction('init', ['if' => 'condition'])
+     * // instead of
+     * $this->addAfterAction([$this, 'init'], ['if' => [$this, 'condition']])
+     * ```
+     *
+     * @param callable|string $callable Callable to be invoked as after action (after filter)
+     * @param array $options Array of options
+     * @return void
+     */
+    public function addAfterAction($callable, $options = [])
+    {
+        $this->on('Controller.shutdown', $callable, $options);
+    }
+
+    /**
+     * Shortcut for binding events on controller's event manager.
+     * See usage of addBeforeAction or addAfterAction methods.
+     *
+     * @param string $event Event name
+     * @param callable|string $callable Callable to be used as an event handler
+     * @param array $options Array of options
+     * @return void
+     */
+    public function on($event, $callable, $options = [])
```

Comment:
;-) the more I think about it ... if we move the method into EventDispatcherTrait, everything that uses it can take advantage of simplified conditional callables.
```php
$table->on('Model.beforeSave', 'doSomething', ['if' => 'isSomethingTrue'])
```
`only` and `except` however only make sense for controllers, so they must stay somewhere here